src/compiler/factory/emitHelpers.ts

```diff
@@ -59,6 +59,8 @@ export interface ESDecorateClassContext {
      * The name of the decorated element.
      */
     name: Expression;
+
+    metadata?: Expression;

```

Comment:
There's no reason for this to be conditional here, it will only be conditional at runtime.